ironic/resource_ironic_deployment.go

```diff
@@ -84,9 +99,43 @@ func resourceDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.SetId(d.Get("node_uuid").(string))
+	// FIXME: only retrieves from URL when a cert is defined
+	// The cert is base64 encoded from the installer
+	userDataURL := d.Get("user_data_url")
+	userData := d.Get("user_data")
+	if userDataURL != nil {
+		caCertPool := x509.NewCertPool()
+		userDataCaCert := d.Get("user_data_url_ca_cert")
+		if userDataCaCert != nil {
+			caCert, err := base64.StdEncoding.DecodeString(userDataCaCert.(string))
+			if err != nil {
+				return fmt.Errorf("could not decode user_data_url_ca_cert: %s", err)
+			}
+			caCertPool.AppendCertsFromPEM(caCert)
+		}
+
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+				RootCAs:      caCertPool,
+				},
+			},
+		}
+
+		// Get the data
+		resp, err := client.Get(userDataURL.(string))
+		if err != nil {
+			return fmt.Errorf("could not get user_data_url: %s", err)
+		}
+		defer resp.Body.Close()
+		userData, err := ioutil.ReadAll(resp.Body)
```

Comment:
Isn't the scope of this userData local to the `if` block. So in the end, aren't we still passing the stub userData rather than the one fetched in the 'if' block?

Reply:
Yes well spotted - I found the same issue running the tests locally, will fix :+1: